File: frontend/src/types/resumeImprovement.ts
export type ResumeImprovementSessionStatus = 'draft' | 'active' | 'completed' | 'archived';
export type ResumeImprovementPriority = 'high' | 'medium' | 'low';
export type ResumeImprovementSuggestionStatus = 'pending' | 'accepted' | 'rejected';

export interface ResumeImprovementSuggestion {
  suggestionId: string;
  sessionId: string;
  section: string;
  originalText: string;
  suggestedText: string;
  reason: string;
  priority: ResumeImprovementPriority;
  status: ResumeImprovementSuggestionStatus;
  createdAt: string;
  updatedAt: string;
}

export interface ResumeImprovementDraft {
  draftId: string;
  sessionId: string;
  resumeId: string;
  sections: Record<string, string[]>;
  createdAt: string;
  updatedAt: string;
}

export interface ResumeImprovementSession {
  sessionId: string;
  userId: string;
  resumeId: string;
  screeningResultId?: string;
  status: ResumeImprovementSessionStatus;
  createdAt: string;
  updatedAt: string;
  completedAt?: string;
  suggestionsCount: number;
  acceptedCount: number;
  rejectedCount: number;
}

export interface GenerateSuggestionRequest {
  section: string;
  context?: string;
}

export interface EditSuggestionRequest {
  editedText: string;
}

export interface ResumeImprovementSummary {
  session: ResumeImprovementSession;
  suggestions: ResumeImprovementSuggestion[];
  draft?: ResumeImprovementDraft;
}

File: backend/app/schemas/resume_improvement.py
from pydantic import BaseModel, Field
from typing import List, Optional, Dict, Any
from enum import Enum
from datetime import datetime

class ResumeImprovementSessionStatus(str, Enum):
    draft = "draft"
    active = "active"
    completed = "completed"
    archived = "archived"

class ResumeImprovementPriority(str, Enum):
    high = "high"
    medium = "medium"
    low = "low"

class ResumeImprovementSuggestionStatus(str, Enum):
    pending = "pending"
    accepted = "accepted"
    rejected = "rejected"

class ResumeImprovementSuggestion(BaseModel):
    suggestionId: str
    sessionId: str
    section: str
    originalText: str
    suggestedText: str
    reason: str
    priority: ResumeImprovementPriority
    status: ResumeImprovementSuggestionStatus
    createdAt: str
    updatedAt: str

class ResumeImprovementDraft(BaseModel):
    draftId: str
    sessionId: str
    resumeId: str
    sections: Dict[str, List[str]] = Field(default_factory=dict)
    createdAt: str
    updatedAt: str

class ResumeImprovementSession(BaseModel):
    sessionId: str
    userId: str
    resumeId: str
    screeningResultId: Optional[str] = None
    status: ResumeImprovementSessionStatus
    createdAt: str
    updatedAt: str
    completedAt: Optional[str] = None
    suggestionsCount: int = 0
    acceptedCount: int = 0
    rejectedCount: int = 0

class GenerateSuggestionRequest(BaseModel):
    section: str
    context: Optional[str] = None

class AcceptSuggestionRequest(BaseModel):
    pass # Empty body for PATCH

class RejectSuggestionRequest(BaseModel):
    pass # Empty body for PATCH

class EditSuggestionRequest(BaseModel):
    editedText: str

class ResumeImprovementSummary(BaseModel):
    session: ResumeImprovementSession
    suggestions: List[ResumeImprovementSuggestion]
    draft: Optional[ResumeImprovementDraft] = None
